python/ray/train/torch.py

```diff
@@ -177,9 +177,15 @@ def __init__(self, base_dataloader: DataLoader, device: torch.device):
         self._dataloader = base_dataloader
         self._device = device
 
+    def _move_to_device(self, item):
+        return (i.to(self.device) for i in item)
```

Comment:
Is `self.device` ever `None`? I can't tell for sure based on the type hints / `__iter__` logic.

Reply:
It shouldn't ever be `None`.

Reply:
Oh in that case can we remove the `None` check in `__iter__`?

Also it looks like we can remove the `device` property and just set `self.device = device` in `__init__` instead?

Reply:
Good point, updated!